compliance/tags/azure_rg_tags/azure_resource_group_tags.pt

```diff
@@ -0,0 +1,314 @@
+  label "Email addresses of the recipients you wish to notify"
+end
+
+parameter "param_azure_tenant_id" do
+  type "string"
+  label "Azure AD Tenant ID"
+  category "Azure"
+end
+
+parameter "param_azure_sub" do
+  type "string"
+  label "Azure Subscription ID"
+  category "Azure"
+end
+
+parameter "param_tag_key" do
+  type "string"
+  label "Tag Key"
+  category "Azure"
+end 
+
+auth "azure_auth", type: "oauth2" do
+  token_url join(["https://login.microsoftonline.com/",$param_azure_tenant_id,"/oauth2/token"])
+  grant type: "client_credentials" do
+    client_id cred("AZURE_APPLICATION_ID")
+    client_secret cred("AZURE_APPLICATION_KEY")
+    additional_params do {
+      "resource" => "https://management.azure.com/"
+    } end
+  end
+end
+
+pagination "azure_pagination" do
+  get_page_marker do
+    body_path "nextLink"
+  end
+  set_page_marker do
+    uri true
+  end
+end
+
+datasource "azure_resources" do
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host "management.azure.com"
+    path join(["/subscriptions/", $param_azure_sub, "/resources"])
+    query "api-version","2018-02-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item,"id")
+      field "region", jmes_path(col_item,"location")
+      field "name", jmes_path(col_item,"name")
+      field "tags", jmes_path(col_item,"tags")
+    end
+  end
+end
+
+datasource "filtered_resources" do
+  run_script $js_filter_resources, $azure_resources, $param_tag_key
+end
+
+script "js_filter_resources", type: "javascript" do
+  parameters "azure_resources", "param_tag_key"
+  result "result"
+  code <<-EOS
+    var result = [];
+
+    // Iterate through all resources
+    _.each(azure_resources, function(ar){
+      console.log('Resource Name: '+ar.name)
+      var id = ar.id ;
+      var rg_split = id.split('/') ;
+      var resource_group_name = rg_split[4] ;
+      var namespace = rg_split[6] ;
+      var resource_type = rg_split[7] ; 
+      console.log('Resource Group Name: '+resource_group_name)
+
+      var resource_tags = ar.tags ;
+      console.log('Tags: '+resource_tags)
+      if (namespace.match(/(Classic)/g)) {
+        // Skip Azure Classic
+        console.log('Skipping Azure Classic resource. ID: '+id)
+      } else {
+        if (_.isEmpty(resource_tags)) {
+          result.push({
+            id: ar["id"],
+            region: ar["region"],
+            name: ar["name"],
+            resource_group: resource_group_name,
+            namespace: namespace,
+            resource_type: resource_type,
+            tag: 'none'
+          })
+        } else {
+          if (_.has(resource_tags, param_tag_key)) {
+            var rg_tag_value = resource_tags[param_tag_key] ;
+            console.log('TAG KEY FOUND')
+            console.log('Tag Value: '+rg_tag_value)
+
+            if (resource_group_name == rg_tag_value) {
```

Comment:
This if/else statement seems redundant as the check is happening in the policy block.

Reply:
yep. was remnant from debugging. good catch.